### ISSUE TITLE
stat: Mean returns Inf for non-infinite inputs

### ISSUE THREAD

**Issue by pkaeding** [gonum/gonum#1048]
### What are you trying to do?
Calculate the mean of high-magnitude input data

### What did you do?
Here is an example of some input values, each fits inside a `float64`, but the sum of them does not. This causes the `Mean` calculation to overflow, and just return `-Inf` (in this case):

In this example, I also offer an alternative mean calculation, using `math/big`.

https://play.golang.org/p/lfpaiMKp9_v

### What did you expect to happen?
Get a non-infinite mean, like `-4.012816463900776e+307`

### What actually happened?
Got `-Inf`

### What version of Go and Gonum are you using?

Go: `go version go1.11.1 darwin/amd64`
gonum: fb5cd163d9249323f95e0808a1e4523ec9d573ea

(but it reproduces on play.golang.org too)

### Does this issue reproduce with the current master?

I haven't tried, but looking at the code, I believe so. The go playground also seemed to download the package when I ran it, so I assume it is using the latest code too.

**Comment by pkaeding** [comment]
I can submit a PR to use `math/big` in the `Mean` implementation if that seems like a good idea. I am just not sure if this project avoids `math/big` on purpose for performance (or any other) reasons?

**Comment by kortschak** [comment]
Yeah, we aren't going to use math/big here. Note that if we implement the running stats package (this is #1037) this will probably no longer be an issue (see https://play.golang.org/p/HoxvbxROnH3)

In the meantime, the Method of Provisional Means can be used in your code with no significant burden.

One must ask though why you have numbers this close to the ceiling and how accurate you expect your calculations to be in this domain.
